api/src/main/java/jakarta/enterprise/concurrent/ContextService.java

```diff
@@ -216,6 +217,45 @@ public interface ContextService {
    */
   public <R> Supplier<R> contextualSupplier(Supplier<R> supplier);
 
+  /**
+   * <p>Wraps a {@link java.util.concurrent.Flow.Subscriber} with context captured from the thread that invokes
+   * <code>contextualSubscriber</code>. Context is captured at the time <code>contextualSubscriber</code> is invoked.</p>
+   *
+   * <p>When one of the methods of the {@link java.util.concurrent.Flow.Subscriber} interface (<code>onSubscribe</code>,
+   * <code>onNext</code>, <code>onError</code>, or <code>onComplete</code>) is invoked on the proxy instance,
+   * context is first established on the thread that will run the method,
+   * then the method of the provided <code>Flow.Subscriber</code> is invoked.
+   * Finally, the previous context is restored on the thread, and the result of the
+   * <code>Flow.Subscriber</code> is returned to the invoker.</p>
+   *
+   * @param <T> the subscribed item type
+   * @param subscriber instance to contextualize
+   * @return contextualized proxy instance that wraps execution of the <code>onSubscribe</code>, <code>onNext</code>,
+   * <code>onError</code>, and <code>onComplete</code> methods
+   * @since 3.1
+   */
+  public <T> Flow.Subscriber<T> contextualSubscriber(Flow.Subscriber<T> subscriber);
+  
+  /**
+   * <p>Wraps a {@link java.util.concurrent.Flow.Processor} with context captured from the thread that invokes
+   * <code>contextualProcessor</code>. Context is captured at the time <code>contextualProcessor</code> is invoked.</p>
+   *
+   * <p>When one of the methods of the {@link java.util.concurrent.Flow.Subscriber} interface (<code>onSubscribe</code>,
+   * <code>onNext</code>, <code>onError</code>, or <code>onComplete</code>) from which {@link java.util.concurrent.Flow.Processor}
+   * extends is invoked on the proxy instance, context is first established on the thread that will run the method,
+   * then the method of the provided <code>Flow.Processor</code> is invoked.
+   * Finally, the previous context is restored on the thread, and the result of the
+   * <code>Flow.Processor</code> is returned to the invoker.</p>
+   *
+   * @param <T> the subscribed item type
+   * @param <R> the published item type
+   * @param processor instance to contextualize
+   * @return contextualized proxy instance that wraps execution of the <code>onSubscribe</code>, <code>onNext</code>,
+   * <code>onError</code>, and <code>onComplete</code> methods
```

Comment:
If making the other related changes,

```suggestion
   * @return contextualized proxy instance that wraps execution of all methods that are inherited from <code>Subscriber</code>.
```